File: binding/src/libqtreewidgetitemiterator.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qtreewidgetitemiterator.html
pub const qtreewidgetitemiterator = struct {

    /// New constructs a new QTreeWidgetItemIterator object.
    pub fn New(it: ?*anyopaque) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_new(@ptrCast(it));
    }


    /// New2 constructs a new QTreeWidgetItemIterator object.
    pub fn New2(widget: ?*anyopaque) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_new2(@ptrCast(widget));
    }


    /// New3 constructs a new QTreeWidgetItemIterator object.
    pub fn New3(item: ?*anyopaque) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_new3(@ptrCast(item));
    }


    /// New4 constructs a new QTreeWidgetItemIterator object.
    pub fn New4(widget: ?*anyopaque, flags: i32) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_new4(@ptrCast(widget), @intCast(flags));
    }


    /// New5 constructs a new QTreeWidgetItemIterator object.
    pub fn New5(item: ?*anyopaque, flags: i32) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_new5(@ptrCast(item), @intCast(flags));
    }


    pub fn OperatorAssign(self: ?*anyopaque, it: ?*anyopaque) void {
        qtc.QTreeWidgetItemIterator_OperatorAssign(@ptrCast(self), @ptrCast(it));
    }

    pub fn OperatorPlusPlus(self: ?*anyopaque, ) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_OperatorPlusPlus(@ptrCast(self));
    }

    pub fn OperatorPlusPlus2(self: ?*anyopaque, param1: i32) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_OperatorPlusPlus2(@ptrCast(self), param1);
    }

    pub fn OperatorPlusAssign(self: ?*anyopaque, n: i32) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_OperatorPlusAssign(@ptrCast(self), n);
    }

    pub fn OperatorMinusMinus(self: ?*anyopaque, ) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_OperatorMinusMinus(@ptrCast(self));
    }

    pub fn OperatorMinusMinus2(self: ?*anyopaque, param1: i32) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_OperatorMinusMinus2(@ptrCast(self), param1);
    }

    pub fn OperatorMinusAssign(self: ?*anyopaque, n: i32) QtC.QTreeWidgetItemIterator {
        return qtc.QTreeWidgetItemIterator_OperatorMinusAssign(@ptrCast(self), n);
    }

    pub fn OperatorMultiply(self: ?*anyopaque, ) QtC.QTreeWidgetItem {
        return qtc.QTreeWidgetItemIterator_OperatorMultiply(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QTreeWidgetItemIterator_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/treewidgetitemiterator.html#types
pub const enums = struct {
    pub const IteratorFlag = enum {
        pub const All: i32 = 0;
        pub const Hidden: i32 = 1;
        pub const NotHidden: i32 = 2;
        pub const Selected: i32 = 4;
        pub const Unselected: i32 = 8;
        pub const Selectable: i32 = 16;
        pub const NotSelectable: i32 = 32;
        pub const DragEnabled: i32 = 64;
        pub const DragDisabled: i32 = 128;
        pub const DropEnabled: i32 = 256;
        pub const DropDisabled: i32 = 512;
        pub const HasChildren: i32 = 1024;
        pub const NoChildren: i32 = 2048;
        pub const Checked: i32 = 4096;
        pub const NotChecked: i32 = 8192;
        pub const Enabled: i32 = 16384;
        pub const Disabled: i32 = 32768;
        pub const Editable: i32 = 65536;
        pub const NotEditable: i32 = 131072;
        pub const UserFlag: i32 = 16777216;
    };

};

File: binding/src/libqtreewidgetitemiterator.cpp
#include <QTreeWidget>
#include <QTreeWidgetItem>
#include <QTreeWidgetItemIterator>
#include <qtreewidgetitemiterator.h>
#include "libqtreewidgetitemiterator.h"
#include "libqtreewidgetitemiterator.hxx"

QTreeWidgetItemIterator* QTreeWidgetItemIterator_new(const QTreeWidgetItemIterator* it) {
	 return new QTreeWidgetItemIterator(*it);
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_new2(QTreeWidget* widget) {
	 return new QTreeWidgetItemIterator(widget);
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_new3(QTreeWidgetItem* item) {
	 return new QTreeWidgetItemIterator(item);
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_new4(QTreeWidget* widget, int flags) {
	 return new QTreeWidgetItemIterator(widget, static_cast<QFlags<QTreeWidgetItemIterator::IteratorFlag>>(flags));
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_new5(QTreeWidgetItem* item, int flags) {
	 return new QTreeWidgetItemIterator(item, static_cast<QFlags<QTreeWidgetItemIterator::IteratorFlag>>(flags));
}

void QTreeWidgetItemIterator_OperatorAssign(QTreeWidgetItemIterator* self, const QTreeWidgetItemIterator* it) {
	self->operator=(*it);
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_OperatorPlusPlus(QTreeWidgetItemIterator* self) {
	return new QTreeWidgetItemIterator(self->operator++());
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_OperatorPlusPlus2(QTreeWidgetItemIterator* self, int param1) {
	return new QTreeWidgetItemIterator(self->operator++(param1));
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_OperatorPlusAssign(QTreeWidgetItemIterator* self, int n) {
	return new QTreeWidgetItemIterator(self->operator+=(n));
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_OperatorMinusMinus(QTreeWidgetItemIterator* self) {
	return new QTreeWidgetItemIterator(self->operator--());
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_OperatorMinusMinus2(QTreeWidgetItemIterator* self, int param1) {
	return new QTreeWidgetItemIterator(self->operator--(param1));
}

QTreeWidgetItemIterator* QTreeWidgetItemIterator_OperatorMinusAssign(QTreeWidgetItemIterator* self, int n) {
	return new QTreeWidgetItemIterator(self->operator-=(n));
}

QTreeWidgetItem* QTreeWidgetItemIterator_OperatorMultiply(const QTreeWidgetItemIterator* self) {
	return self->operator*();
}

void QTreeWidgetItemIterator_Delete(QTreeWidgetItemIterator* self) {
    delete self;
}

